src/app/store.ts

```diff
@@ -0,0 +1,35 @@
+import fs from 'fs';
+
+import {jsonStoreFileName} from './constants';
+
+type AppStore = {
+    bot_access_token: string;
+    oauth2_client_secret: string;
+}
+
+class Store {
+    store: AppStore;
+
+    constructor() {
+        this.store = {
+            bot_access_token: '',
+            oauth2_client_secret: '',
+        };
+
+        if (fs.existsSync(jsonStoreFileName)) {
+            fs.readFile(jsonStoreFileName, (err, data) => {
```

Comment:
Seems like we should be saving the store state with this class as well, instead of just having read functionality here. Any store-related `fs` library access should be in this file only 2/5.

Furthermore, we can have an `interface Store` like:

```ts
interface Store {
	getBotAccessToken(): string;
}
```

Then have an implementation like:

```ts
class JSONFileStore implements Store {
	...
}
```

Reply:
https://github.com/mattermost/mattermost-app-zendesk/commit/fa8a3e3c7bcebec08dd689fbae3927ceb05f7847